### ISSUE TITLE
rabbitmq_{user|vhost}.present in test=True reports unnecessary changes

### ISSUE THREAD

**Issue by syphernl** [saltstack/salt#25363]
If you run `salt-call state.highstate test=True` and the state contains a `rabbitmq_user.present` it always results in an item marked as "to be changed":
```
----------
          ID: rabbitmq_user_userxyz
    Function: rabbitmq_user.present
        Name: userxyz
      Result: None
     Comment: User userxyz already presents
     Started: 10:10:31.939836
    Duration: 392.501 ms
     Changes:   
```

Most likely the `result: None` should be changed to `result: True` in case the user already exists. The same is the case for other states (e.g. `service.running`)

The same applies to `rabbitmq_vhost.present` states which also return `None` rather than `True`

**Comment by jfindlay** [comment]
@syphernl, thanks for the report.

This is nearly a duplicate of #24856, but has more information on `rabbitmq_vhost.present` also behaving incorrectly under `test=True`.

**Comment by rallytime** [comment]
@syphernl I've refactored the rabbitmq_user.present state to make sure we're reporting errors and test=true functionality correctly. Can you give that a spin and let me know if you run into any problems?

The vhost.present state still needs to be fixed, however.

**Comment by rallytime** [comment]
I have fixed the rabbitmq_vhost states in #30567 so that test=true reports `None` only when there are changes.
